src/shared/user-record.c

```diff
@@ -2252,6 +2252,159 @@ const char** user_record_self_modifiable_privileged(UserRecord *h) {
         return (const char**) h->self_modifiable_privileged ?: (const char**) default_fields;
 }
 
+static int remove_self_modifiable_json_fields_common(UserRecord *current, sd_json_variant **ret) {
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL, *blobs = NULL;
+        char **allowed;
+        int r;
+
+        assert(current);
+        assert(ret);
+
+        if (!sd_json_variant_is_object(*ret))
```

Comment:
in our coding style return parameters should be called `ret` or `ret_xyz`, no other arguments. But here you are dereferencing what is passed in, which means this is not purely a return param anymore. hence shouldn't be called "ret".